nifi-extension-bundles/nifi-network-bundle/nifi-network-processors/src/main/java/org/apache/nifi/processors/network/pcap/SplitPCAP.java

```diff
@@ -0,0 +1,237 @@
+            .build();
+    public static final Relationship REL_SPLIT = new Relationship.Builder()
+            .name("split")
+            .description("The individual PCAP 'segments' of the original PCAP FlowFile will be routed to this relationship.")
+            .build();
+
+    private static final List<PropertyDescriptor> DESCRIPTORS = List.of(PCAP_MAX_SIZE);
+    private static final Set<Relationship> RELATIONSHIPS = Set.of(REL_ORIGINAL, REL_FAILURE, REL_SPLIT);
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return RELATIONSHIPS;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return DESCRIPTORS;
+    }
+
+    /**
+     * This method is called when a trigger event occurs in the processor.
+     * It processes the incoming flow file, splits it into smaller pcap files based on the PCAP Max Size,
+     * and transfers the split pcap files to the success relationship.
+     * If the flow file is empty or not parseable, it is transferred to the failure relationship.
+     *
+     * @param context  the process context
+     * @param session  the process session
+     */
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) {
+
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final List<FlowFile> splitFiles = new ArrayList<>();
+
+        try {
+            session.read(flowFile, new InputStreamCallback() {
+                @Override
+                public void process(final InputStream inStream) throws IOException {
+                    final int pcapMaxSize = context.getProperty(PCAP_MAX_SIZE.getName()).asDataSize(DataUnit.B).intValue();
+                    final List<Packet> loadedPackets = new ArrayList<>();
+                    final BufferedInputStream bInStream = new BufferedInputStream(inStream);
+                    int totalPackets = 0;
+
+                    if ( bInStream.available() == 0 ) {
+                        throw new ProcessException("PCAP file empty.");
+                    }
+
+                    final byte[] pcapHeader = new byte[PCAP.PCAP_HEADER_LENGTH];
+                    bInStream.read(pcapHeader, 0, PCAP.PCAP_HEADER_LENGTH);
+                    int currentPcapTotalLength = PCAP.PCAP_HEADER_LENGTH;
+
+                    final PCAP templatePcap = new PCAP(new ByteBufferInterface(pcapHeader));
+
+                    while (bInStream.available() > 0) {
+
+                        byte[] packetHeader = new byte[Packet.PACKET_HEADER_LENGTH];
+                        bInStream.read(packetHeader, 0, Packet.PACKET_HEADER_LENGTH);
+                        Packet currentPacket = new Packet(packetHeader, templatePcap);
+
+                        if (currentPacket.totalLength() > pcapMaxSize) {
+                            throw new ProcessException("PCAP contains a packet larger than the max size.");
```

Comment:
Recommend including the packet length in the message:
```suggestion
                            throw new ProcessException("PCAP Packet length [%d] larger then configured maximum [%d]".formatted(currentPacket.totalLength(), pcapMaxSize));
```